File: dhkim/10866.rs
use std::collections::LinkedList;
use std::io;
use std::fmt::Write;
use std::str::SplitWhitespace;

fn op(cmd: &mut SplitWhitespace, deque: &mut LinkedList<usize>) -> String {
    let mut output = String::new();
    match cmd.next().unwrap() {
        "push_front" => {
            deque.push_front(cmd.next().unwrap().parse::<usize>().unwrap())
        },
        "push_back" => {
            deque.push_back(cmd.next().unwrap().parse::<usize>().unwrap())
        },
        "pop_front" => {
            match deque.pop_front() {
                Some(i) => {
                    writeln!(output, "{i}").unwrap();
                },
                None => {
                    writeln!(output, "{}", -1).unwrap();
                }
            }
        },
        "pop_back" => {
            match deque.pop_back() {
                Some(i) => {
                    writeln!(output, "{i}").unwrap();
                },
                None => {
                    writeln!(output, "{}", -1).unwrap();
                }
            }
        },
        "size" => {
            writeln!(output, "{}", deque.len()).unwrap();
        },
        "empty" => {
            writeln!(output, "{}", if deque.is_empty() { 1 } else { 0 }).unwrap();
        },
        "front" => {
            match deque.front() {
                Some(i) => {
                    writeln!(output, "{i}").unwrap();
                },
                None => {
                    writeln!(output, "{}", -1).unwrap();
                }
            }
        },
        "back" => {
            match deque.back() {
                Some(i) => {
                    writeln!(output, "{i}").unwrap();
                },
                None => {
                    writeln!(output, "{}", -1).unwrap();
                }
            }
        },
        _ => { }
    }

    output
}

fn main() {
    let mut input = String::new();
    let mut output = String::new();

    io::stdin().read_line(&mut input).unwrap();
    let n: usize = input.trim().parse().unwrap();

    let mut deque = LinkedList::new();
    for _ in 0..n {
        input.clear();
        io::stdin().read_line(&mut input).unwrap();
        let mut cmd = input.split_whitespace();
        write!(output, "{}", op(&mut cmd, &mut deque)).unwrap();
    }

    println!("{output}");
}

File: medAndro/10866.py
import sys
from collections import deque

mydeque = deque()

def parse_txt(str):
    if str[0] == 'push_front':
        mydeque.appendleft(str[1])
    elif str[0] == 'push_back':
        mydeque.append(str[1])
    elif str[0] == 'pop_front':
        if mydeque:
            print(mydeque.popleft())
        else:
            print(-1)
    elif str[0] == 'pop_back':
        if mydeque:
            print(mydeque.pop())
        else:
            print(-1)
    elif str[0] == 'size':
        print(len(mydeque))
    elif str[0] == 'empty':
        print(int(not bool(mydeque)))
    elif str[0] == 'front':
        if mydeque:
            print(mydeque[0])
        else:
            print(-1)
    elif str[0] == 'back':
        if mydeque:
            print(mydeque[-1])
        else:
            print(-1)

num = int(sys.stdin.readline().rstrip())

for i in range(num):
    parse_txt(sys.stdin.readline().split())